lib/addons/ebs-csi-driver/index.ts

```diff
@@ -1,49 +1,142 @@
-        super({
-            addOnName: defaultProps.addOnName,
-            version: options?.version ?? defaultProps.version,
-            saName: defaultProps.saName
-        });
-    }
+  constructor(readonly options?: EbsCsiDriverAddOnProps) {
+    super({
+      addOnName: defaultProps.addOnName,
+      version: options?.version ?? defaultProps.version,
+      saName: defaultProps.saName,
+    });
+
+    this.ebsProps = {
+      ...defaultProps,
+      ...options,
+    };
+  }
+
+  providePolicyDocument(clusterInfo: ClusterInfo): PolicyDocument {
+    return getEbsDriverPolicyDocument(
+      clusterInfo.cluster.stack.partition,
+      this.options?.kmsKeys
+    );
+  }
+
+  async deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    const baseDeployment = await super.deploy(clusterInfo);
+
+    const cluster = clusterInfo.cluster;
+    let updateSc: KubernetesManifest;
+
+    if (this.ebsProps.storageClass) {
+      // patch resource on cluster
+    const patchSc =  new KubernetesPatch(cluster.stack, `${cluster}-RemoveGP2SC`, {
+        cluster: cluster,
+        resourceName: "storageclass/gp2",
+        applyPatch: {
+          metadata: {
+            annotations: {
+              "storageclass.kubernetes.io/is-default-class": "false",
+            },
+          },
+        },
+        restorePatch: {
+          metadata: {
+            annotations: {
+              "storageclass.kubernetes.io/is-default-class": "true",
+            },
+          },
+        },
+      });
 
-    providePolicyDocument(clusterInfo: ClusterInfo) : PolicyDocument {
-        return getEbsDriverPolicyDocument(clusterInfo.cluster.stack.partition, this.options?.kmsKeys);
+      // Create and set gp3 StorageClass as cluster-wide default
+      updateSc = new KubernetesManifest(
+        cluster.stack,
+        `${cluster}-SetDefaultSC`,
+        {
+          cluster: cluster,
+          manifest: [
+            {
+              apiVersion: "storage.k8s.io/v1",
+              kind: "StorageClass",
+              metadata: {
+                name: "gp3",
+                annotations: {
+                  "storageclass.kubernetes.io/is-default-class": "true",
+                },
+              },
+              provisioner: "ebs.csi.aws.com",
+              reclaimPolicy: "Delete",
+              volumeBindingMode: "WaitForFirstConsumer",
+              parameters: {
+                type: "gp3",
+                fsType: "ext4",
+                encrypted: "true",
+              },
+            },
+          ],
+        }
+      );
+
+      /* new scConstruct(cluster.stack, `${cluster}-PatchSC`, updateSc);
+      return Promise.resolve(updateSc);
+    } else {
+      const noOpConstruct = new Construct(cluster.stack, `${cluster}-NoOp`);
+      return Promise.resolve(noOpConstruct); */
+      patchSc.node.addDependency(baseDeployment);
+      updateSc.node.addDependency(baseDeployment);
+
+      return baseDeployment;

```

Comment:
when you return baseDeployment that means that csi driver is installed but patch and update are not done yet. 
I believe customers will want to rely on the SC to be installed so they can depend on the correct SC used in their addon. 
So to address this, the following should be done:

1. `updateSc` should depend on `patchSc` (which in turn depends on the baseDeployment). 
2. Return `updateSc` - this will make sure baseDeployment, patchSc and update are all done.

Reply:
Updated as per this!